### PR TITLE
Introduce DALROOT and Major Minor versions in env scripts

### PR DESCRIPTION
**Pull request by napetrov**
# Description
Adding DALROOT variable
Adding Major and Minor version env variables to identify DLL versions
Updating path for installed lib search

### CODE REVIEW

**Comment by rlnx** [comment]
Shouldn't it be 'ONEDALROOT'

**Comment by napetrov** [comment]
@rlnx  - on system we are instilled in 'dal' dir to align with other components. 
From variable name - short variant is also aligned with other libs

So the main thought was to align with other components

File: deploy/local/dal [hunk 1]
```diff
@@ -84,6 +84,9 @@ set daal_target_arch         "intel64"
 module-whatis "Intel(R) Data Analytics Library for $daal_target_arch."
```
Comment (rlnx) [inline deploy/local/dal:84]: Shouldn't it be official product name?

```suggestion
module-whatis "Intel(R) oneAPI Data Analytics Library for $daal_target_arch."
```

File: deploy/local/vars_win.bat [hunk 2]
```diff
@@ -52,6 +52,9 @@ if exist "%DAAL_UP_OLD%\redist" (
 )
 set "CLASSPATH=%DAAL%\lib\onedal.jar;%CLASSPATH%"
 endlocal& ^
+set DAL_MAJOR_BINARY=1& ^
+set DAL_MINOR_BINARY=0& ^
+set DALROOT=%DAALROOT%& ^
```
Comment (rlnx) [inline deploy/local/vars_win.bat:57]: I see section `:GoodArgs` at line 43, don't we need to add `DAL_MAJOR`/`DAL_MINOR` there?

File: deploy/local/dal [hunk 3]
```diff
@@ -84,6 +84,9 @@ set daal_target_arch         "intel64"
...
 
 # Setup environment variables
+setenv          DAL_MAJOR_BINARY   1
+setenv          DAL_MINOR_BINARY   0
```
Comment (averbukh) [inline deploy/local/dal:88]: Much better is to specify values for variables in separate text file in following form:
var1=value1
var2=value2
and use this file to generate code to set environment variables for every OS.

For example, on Linux to do so, you can use sed/awk/... to perform required transformations and use `source <(cat version.txt | <transformations>)` to do actual set of env.variables.

Comment (napetrov) [reply deploy/local/dal:88]: yes, and ideally this should be taken from single file with version for entire product and not separate version.txt file. 

But this is work for @Alexsandruss that will be completed after release .  For now i would like to expose variables for users in env

File: deploy/local/vars_win.bat (hunk 2 above)
Comment (napetrov) [reply deploy/local/vars_win.bat:57]: Thins within setlocal context is get used for different data manipulations and calculations but actual values are set after endlocal section.  As far we have no processing for now - it's sufficient to specify them at the end. 

but as this have to be moved to version parsing - this logic will be moved to setlocal context and GoodArgs point

File: deploy/local/dal (hunk 3 above)
Comment (averbukh) [reply deploy/local/dal:88]: OK

**Review by averbukh — APPROVED** [review]

**Pull request merged**